timelock-agent/src/main/java/com/palantir/timelock/paxos/PersistedSchemaVersion.java

```diff
@@ -53,14 +56,21 @@ void upgradeVersion(long targetVersion) {
         execute(dao -> {
             if (dao.getVersion(ONLY_ROW).orElse(0L) < targetVersion) {
                 dao.setVersion(ONLY_ROW, targetVersion);
+                persistedVersion = Optional.of(targetVersion);
```

Comment:
note to reader: this looks funky but is actually correct - execute means execute with a connection, not in one transaction